spot-vaults/contracts/UsdcSpotManager.sol

```diff
@@ -0,0 +1,228 @@
+    /// @dev Token Constants.
+    uint256 public constant ONE_SPOT = 1e9;
+    uint256 public constant ONE_USDC = 1e6;
+
+    /// @dev Decimals.
+    uint256 public constant DECIMALS = 18;
+    uint256 public constant ONE = (10 ** DECIMALS);
+
+    /// @dev We bound the deviation factor to 100.0.
+    uint256 public constant MAX_DEVIATION = 100 * ONE; // 100.0
+
+    //-------------------------------------------------------------------------
+    // Storage
+
+    /// @notice The USDC-SPOT charm alpha vault.
+    IAlphaProVault public immutable VAULT;
+
+    /// @notice The underlying USDC-SPOT univ3 pool.
+    IUniswapV3Pool public immutable POOL;
+
+    /// @notice The vault's token0, the USDC token.
+    address public immutable USDC;
+
+    /// @notice The vault's token1, the SPOT token.
+    address public immutable SPOT;
+
+    /// @notice Pricing strategy to price the SPOT token.
+    IBillBrokerPricingStrategy public spotAppraiser;
+
+    /// @notice The contract owner.
+    address public owner;
+
+    //-------------------------------------------------------------------------
+    // Limit range calculation parameters
+    //
+    // The deviation factor as calculated by the vault is the ratio of
+    // SPOT's market price to it's FMV price. When the deviation factor = ONE,
+    // SPOT is trading at its FMV price.
+    //
+    // Based on the deviation factor the manager selectively updates
+    // the vault's limit threshold values.
+    //
+    // If the deviation is above ONE (ie SPOT is trading at a premium) and the vault is SPOT heavy,
+    // Or if the deviation is below ONE (ie SPOT is trading at a discount) and the vault is USDC heavy,
+    // vault chooses a narrow limit range.
+    // Otherwise, the vault defaults to a wide limit range.
+    //
+    // This strategy attempts to reduce IL using knowledge about the behavior of SPOT.
+    // Put simply, The vault tries NOT to sell SPOT below FMV or buy SPOT above FMV.
+    //
+
+    /// @notice The narrow limit range, in univ3 ticks.
+    int24 public limitThresholdNarrow;
+
+    /// @notice The default wide limit range, in univ3 ticks.
+    int24 public limitThresholdWide;
+
+    //--------------------------------------------------------------------------
+    // Modifiers
+
+    modifier onlyOwner() {
+        // solhint-disable-next-line custom-errors
+        require(msg.sender == owner, "Unauthorized caller");
+        _;
+    }
+
+    //-----------------------------------------------------------------------------
+    // Constructor and Initializer
+
+    /// @notice Constructor initializes the contract with provided addresses.
+    /// @param vault_ Address of the AlphaProVault contract.
+    /// @param spotAppraiser_ Address of the spot appraiser.
+    constructor(IAlphaProVault vault_, IBillBrokerPricingStrategy spotAppraiser_) {
+        owner = msg.sender;
+
+        VAULT = vault_;
+        POOL = vault_.pool();
+        USDC = vault_.token0();
```

Comment:
Probably better style to hardcode the expected token addresses, then do a `require` check on the vault properties on init.

Nothing inherently incorrect with this approach, provided we do a manual spot check on deployment.